draft-irtf-cfrg-sigma-protocols.md

```diff
@@ -91,30 +88,97 @@ informative:
 
 --- abstract
 
-This document describes interactive sigma protocols, a class of secure, general-purpose zero-knowledge proofs of knowledge consisting of three moves: commitment, challenge, and response. Concretely, the protocol allows one to prove knowledge of a secret witness without revealing any information about it.
+This document describes Sigma protocols, a family of interactive zero-knowledge proofs that enable a prover to convince a verifier of the validity of a statement without revealing any additional information.
+A Sigma protocol proceeds in three moves—commitment, challenge, and response—and provides the foundational structure for many cryptographic constructions, including digital signatures, identification schemes, and complex zero-knowledge proofs.
+Concretely, Sigma protocols allow proving knowledge of secret values satisfying certain relations, such as discrete logarithm equations, while preserving zero-knowledge and soundness.
+Being zero-knowledge, they ensure that no information about the witness is leaked, while the verifier remains assured of the correctness of the claimed statement.
 
 --- middle
 
-# Introduction
+# 1. Introduction
 
-Any sigma protocol must define three objects: a *commitment* (computed by the prover), a *challenge* (computed by the verifier), and a *response* (computed by the prover).
+A *Sigma protocol* is a three-move interactive proof system between two parties: a Prover, who holds a secret witness, and a Verifier, who wishes to be convinced that the Prover knows a valid witness for a given statement.
+The protocol proceeds through three sequential steps—commitment, challenge, and response—that together enable the Verifier to check the validity of the claim without learning anything about the witness itself.
 
-## Core interface
+During execution, the Prover first sends a *commitment* to the Verifier, encapsulating information derived from the witness without revealing it. The Verifier then issues a random *challenge*, ensuring that the proof cannot be reused or forged. Finally, the Prover computes a *response* based on both the challenge and the witness, which the Verifier uses to verify correctness.
 
-The public functions are obtained relying on an internal structure containing the definition of a sigma protocol.
+The following figure illustrates the general message flow of a Sigma protocol, highlighting the interactive nature of the exchange between the Prover and the Verifier:
+
+~~~ ascii-art
+  (1) prover_commit()                     (6) verifier()
```

Comment:
what is this picture meant to describe? More precisely:
what is the meaning of the self-feeding arrows? and the one of the verifier? 
Maybe you can use the picture from here: eprint.iacr.org/2025/536.pdf and just pick the interactive protocol?